### PR TITLE
fix glooctl assets arch

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Build glooctl assets with correct arch

# Testing
`make build-cli` and run `file` on each output bin in `_output/glooctl*`

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/9295

### CODE REVIEW

File: Makefile.ci [hunk 1]
```diff
@@ -10,17 +10,17 @@ CLI_DIR=projects/gloo/cli
...
 $(OUTPUT_DIR)/glooctl-darwin-amd64: $(SOURCES)
-	$(GO_BUILD_FLAGS) GOOS=darwin go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
+	$(GO_BUILD_FLAGS) GOOS=darwin GOARCH=amd64 go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
 $(OUTPUT_DIR)/glooctl-darwin-arm64: $(SOURCES)
-	$(GO_BUILD_FLAGS) GOOS=darwin go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
+	$(GO_BUILD_FLAGS) GOOS=darwin GOARCH=arm64 go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
 
 $(OUTPUT_DIR)/glooctl-windows-$(GOARCH).exe: $(SOURCES)
-	$(GO_BUILD_FLAGS) GOOS=windows go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
+	$(GO_BUILD_FLAGS) GOOS=windows GOARCH=$(GOARCH) go build -ldflags=$(LDFLAGS) -gcflags=$(GCFLAGS) -o $@ $(CLI_DIR)/cmd/main.go $(STDERR_SILENCE_REDIRECT)
```
Comment (ben-taussig-solo) [inline Makefile.ci:23]: It looks like we only release an amd64 variant of the windows build of glooctl
https://github.com/solo-io/gloo/blob/20d600038c0b8f42394f71fc8544349d70f43602/ci/upload_github_release_assets.go#L59-L63

How do you feel about making this 
`$(OUTPUT_DIR)/glooctl-windows-amd64.exe: $(SOURCES)`

I don't think it's possible that anyone was using the prior make target to build for other architectures, as we weren't actually passing the `GOARCH` env var through

Comment (jbohanon) [reply Makefile.ci:23]: I don't see the harm in leaving it to keep the option open, but I'm happy to change it

Comment (ben-taussig-solo) [reply Makefile.ci:23]: Yeah now that I look at it, I don't think the glooctl binary should have many issues compiling to different targets. I think let's just leave it as-is

**Review by ben-taussig-solo — APPROVED** [review]
- I was able to compile all of the linux and darwin targets on darwin/arm64.
 - I was also able to compile windows/arm64 -- I didn't attempt to compile windows for other targets
 - darwin/amd64 seems to run properly via emulation, and darwin/arm64 seems to run properly natively. 
 - linux/amd64 runs properly on the ubuntu VM I use for envoy dev
 - I have not tested linux/arm64 or windows/arm64, but I have no reason to believe they don't run as expected

**Review by sam-heilbron — APPROVED** [review]
🚀

**Pull request merged**